server/container/core/src/test/java/org/apache/james/core/MimeMessageCopyOnWriteProxyTest.java

```diff
@@ -29,96 +31,103 @@
+        MimeMessageCopyOnWriteProxy messageFromSources = (MimeMessageCopyOnWriteProxy) getMessageFromSources(content + sep + body);
+        MailImpl mail = new MailImpl("test", new MailAddress("test@test.com"), recipientList, messageFromSources);
+        MailImpl mail2 = (MailImpl) mail.duplicate();
+
+        assertThat(mail.getMessage()).isNotSameAs(mail2.getMessage());
     }
 
     @Test
-    public void testMessageCloning2() throws Exception {
-        ArrayList<MailAddress> r = new ArrayList<MailAddress>();
-        r.add(new MailAddress("recipient@test.com"));
-        MimeMessageCopyOnWriteProxy messageFromSources = (MimeMessageCopyOnWriteProxy) getMessageFromSources(
-                content + sep + body);
-        MailImpl mail = new MailImpl("test", new MailAddress("test@test.com"), r, messageFromSources);
-        MailImpl m2 = (MailImpl) mail.duplicate();
-        System.out.println("mail: " + getReferences(mail.getMessage()) + " m2: " + getReferences(m2.getMessage()));
-        assertNotSame(m2, mail);
-        assertNotSame(m2.getMessage(), mail.getMessage());
-        // test that the wrapped message is the same
-        assertTrue(isSameMimeMessage(m2.getMessage(), mail.getMessage()));
-        // test it is the same after real only operations!
-        m2.getMessage().getSubject();
-        assertTrue(isSameMimeMessage(m2.getMessage(), mail.getMessage()));
-        m2.getMessage().setText("new body");
-        m2.getMessage().saveChanges();
-        // test it is different after a write operation!
-        m2.getMessage().setSubject("new Subject");
-        assertTrue(!isSameMimeMessage(m2.getMessage(), mail.getMessage()));
-        // check that the subjects are correct on both mails!
-        assertEquals(m2.getMessage().getSubject(), "new Subject");
-        assertEquals(mail.getMessage().getSubject(), "foo");
-        // cloning again the messages
-        Mail m2clone = m2.duplicate();
-        assertTrue(isSameMimeMessage(m2clone.getMessage(), m2.getMessage()));
-        MimeMessage mm = getWrappedMessage(m2.getMessage());
-        assertNotSame(m2.getMessage(), m2clone.getMessage());
-        // test that m2clone has a valid wrapped message
-        MimeMessage mm3 = getWrappedMessage(m2clone.getMessage());
-        assertNotNull(mm3);
-        // dispose m2 and check that the clone has still a valid message and it
-        // is the same!
-        LifecycleUtil.dispose(m2);
-        assertEquals(mm3, getWrappedMessage(m2clone.getMessage()));
-        // change the message that should be not referenced by m2 that has
-        // been disposed, so it should not clone it!
-        m2clone.getMessage().setSubject("new Subject 2");
-        m2clone.getMessage().setText("new Body 3");
-        assertTrue(isSameMimeMessage(m2clone.getMessage(), mm));
-        LifecycleUtil.dispose(mail);
-        LifecycleUtil.dispose(messageFromSources);
+    public void mailShouldWrapMimeMessage() throws Exception {
```

Comment:
Using a MimeMessageWrapper for cloning returns a different message. So it is kind of implied. I will make it more readable.